rewrite-core/src/main/java/org/openrewrite/trait/JavaTypeReference.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.trait;
+
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.Tree;
+
+public interface JavaTypeReference extends Trait<Tree> {
```

Comment:
Should I remove the `extends` here and place it back on `SpringJavaTypeReference.java` as to not force a `JavaTypeReference` to be a trait?